// @aws-sdk/client-comprehendmedical/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationException = exports.SNOMEDCTEntityType = exports.SNOMEDCTAttributeType = exports.SNOMEDCTTraitName = exports.SNOMEDCTRelationshipType = exports.SNOMEDCTEntityCategory = exports.RxNormEntityType = exports.RxNormEntityCategory = exports.RxNormAttributeType = exports.RxNormTraitName = exports.ICD10CMEntityCategory = exports.ICD10CMAttributeType = exports.ICD10CMTraitName = exports.ICD10CMRelationshipType = exports.ICD10CMEntityType = exports.TextSizeLimitExceededException = exports.ServiceUnavailableException = exports.InvalidEncodingException = exports.TooManyRequestsException = exports.ResourceNotFoundException = exports.InvalidRequestException = exports.InternalServerException = exports.LanguageCode = exports.JobStatus = exports.EntitySubType = exports.AttributeName = exports.RelationshipType = exports.EntityType = void 0;
const ComprehendMedicalServiceException_1 = require("./ComprehendMedicalServiceException");
exports.EntityType = {
    ANATOMY: "ANATOMY",
    BEHAVIORAL_ENVIRONMENTAL_SOCIAL: "BEHAVIORAL_ENVIRONMENTAL_SOCIAL",
    MEDICAL_CONDITION: "MEDICAL_CONDITION",
    MEDICATION: "MEDICATION",
    PROTECTED_HEALTH_INFORMATION: "PROTECTED_HEALTH_INFORMATION",
    TEST_TREATMENT_PROCEDURE: "TEST_TREATMENT_PROCEDURE",
    TIME_EXPRESSION: "TIME_EXPRESSION",
};
exports.RelationshipType = {
    ACUITY: "ACUITY",
    ADMINISTERED_VIA: "ADMINISTERED_VIA",
    AMOUNT: "AMOUNT",
    DIRECTION: "DIRECTION",
    DOSAGE: "DOSAGE",
    DURATION: "DURATION",
    EVERY: "EVERY",
    FOR: "FOR",
    FORM: "FORM",
    FREQUENCY: "FREQUENCY",
    NEGATIVE: "NEGATIVE",
    OVERLAP: "OVERLAP",
    QUALITY: "QUALITY",
    RATE: "RATE",
    ROUTE_OR_MODE: "ROUTE_OR_MODE",
    STRENGTH: "STRENGTH",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
    TEST_UNIT: "TEST_UNIT",
    TEST_UNITS: "TEST_UNITS",
    TEST_VALUE: "TEST_VALUE",
    USAGE: "USAGE",
    WITH_DOSAGE: "WITH_DOSAGE",
};
exports.AttributeName = {
    DIAGNOSIS: "DIAGNOSIS",
    FUTURE: "FUTURE",
    HYPOTHETICAL: "HYPOTHETICAL",
    LOW_CONFIDENCE: "LOW_CONFIDENCE",
    NEGATION: "NEGATION",
    PAST_HISTORY: "PAST_HISTORY",
    PERTAINS_TO_FAMILY: "PERTAINS_TO_FAMILY",
    SIGN: "SIGN",
    SYMPTOM: "SYMPTOM",
};
exports.EntitySubType = {
    ACUITY: "ACUITY",
    ADDRESS: "ADDRESS",
    AGE: "AGE",
    ALCOHOL_CONSUMPTION: "ALCOHOL_CONSUMPTION",
    ALLERGIES: "ALLERGIES",
    AMOUNT: "AMOUNT",
    BRAND_NAME: "BRAND_NAME",
    CONTACT_POINT: "CONTACT_POINT",
    DATE: "DATE",
    DIRECTION: "DIRECTION",
    DOSAGE: "DOSAGE",
    DURATION: "DURATION",
    DX_NAME: "DX_NAME",
    EMAIL: "EMAIL",
    FORM: "FORM",
    FREQUENCY: "FREQUENCY",
    GENDER: "GENDER",
    GENERIC_NAME: "GENERIC_NAME",
    ID: "ID",
    IDENTIFIER: "IDENTIFIER",
    NAME: "NAME",
    PHONE_OR_FAX: "PHONE_OR_FAX",
    PROCEDURE_NAME: "PROCEDURE_NAME",
    PROFESSION: "PROFESSION",
    QUALITY: "QUALITY",
    QUANTITY: "QUANTITY",
    RACE_ETHNICITY: "RACE_ETHNICITY",
    RATE: "RATE",
    REC_DRUG_USE: "REC_DRUG_USE",
    ROUTE_OR_MODE: "ROUTE_OR_MODE",
    STRENGTH: "STRENGTH",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
    TEST_NAME: "TEST_NAME",
    TEST_UNIT: "TEST_UNIT",
    TEST_UNITS: "TEST_UNITS",
    TEST_VALUE: "TEST_VALUE",
    TIME_EXPRESSION: "TIME_EXPRESSION",
    TIME_TO_DX_NAME: "TIME_TO_DX_NAME",
    TIME_TO_MEDICATION_NAME: "TIME_TO_MEDICATION_NAME",
    TIME_TO_PROCEDURE_NAME: "TIME_TO_PROCEDURE_NAME",
    TIME_TO_TEST_NAME: "TIME_TO_TEST_NAME",
    TIME_TO_TREATMENT_NAME: "TIME_TO_TREATMENT_NAME",
    TOBACCO_USE: "TOBACCO_USE",
    TREATMENT_NAME: "TREATMENT_NAME",
    URL: "URL",
};
exports.JobStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PARTIAL_SUCCESS: "PARTIAL_SUCCESS",
    STOPPED: "STOPPED",
    STOP_REQUESTED: "STOP_REQUESTED",
    SUBMITTED: "SUBMITTED",
};
exports.LanguageCode = {
    EN: "en",
};
class InternalServerException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerException = InternalServerException;
class InvalidRequestException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRequestException = InvalidRequestException;
class ResourceNotFoundException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class TooManyRequestsException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRequestsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
class InvalidEncodingException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "InvalidEncodingException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidEncodingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidEncodingException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidEncodingException = InvalidEncodingException;
class ServiceUnavailableException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceUnavailableException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceUnavailableException = ServiceUnavailableException;
class TextSizeLimitExceededException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "TextSizeLimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "TextSizeLimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TextSizeLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.TextSizeLimitExceededException = TextSizeLimitExceededException;
exports.ICD10CMEntityType = {
    DX_NAME: "DX_NAME",
    TIME_EXPRESSION: "TIME_EXPRESSION",
};
exports.ICD10CMRelationshipType = {
    OVERLAP: "OVERLAP",
    QUALITY: "QUALITY",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
};
exports.ICD10CMTraitName = {
    DIAGNOSIS: "DIAGNOSIS",
    HYPOTHETICAL: "HYPOTHETICAL",
    LOW_CONFIDENCE: "LOW_CONFIDENCE",
    NEGATION: "NEGATION",
    PERTAINS_TO_FAMILY: "PERTAINS_TO_FAMILY",
    SIGN: "SIGN",
    SYMPTOM: "SYMPTOM",
};
exports.ICD10CMAttributeType = {
    ACUITY: "ACUITY",
    DIRECTION: "DIRECTION",
    QUALITY: "QUALITY",
    QUANTITY: "QUANTITY",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
    TIME_EXPRESSION: "TIME_EXPRESSION",
    TIME_TO_DX_NAME: "TIME_TO_DX_NAME",
};
exports.ICD10CMEntityCategory = {
    MEDICAL_CONDITION: "MEDICAL_CONDITION",
};
exports.RxNormTraitName = {
    NEGATION: "NEGATION",
    PAST_HISTORY: "PAST_HISTORY",
};
exports.RxNormAttributeType = {
    DOSAGE: "DOSAGE",
    DURATION: "DURATION",
    FORM: "FORM",
    FREQUENCY: "FREQUENCY",
    RATE: "RATE",
    ROUTE_OR_MODE: "ROUTE_OR_MODE",
    STRENGTH: "STRENGTH",
};
exports.RxNormEntityCategory = {
    MEDICATION: "MEDICATION",
};
exports.RxNormEntityType = {
    BRAND_NAME: "BRAND_NAME",
    GENERIC_NAME: "GENERIC_NAME",
};
exports.SNOMEDCTEntityCategory = {
    ANATOMY: "ANATOMY",
    MEDICAL_CONDITION: "MEDICAL_CONDITION",
    TEST_TREATMENT_PROCEDURE: "TEST_TREATMENT_PROCEDURE",
};
exports.SNOMEDCTRelationshipType = {
    ACUITY: "ACUITY",
    DIRECTION: "DIRECTION",
    QUALITY: "QUALITY",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
    TEST_UNIT: "TEST_UNIT",
    TEST_UNITS: "TEST_UNITS",
    TEST_VALUE: "TEST_VALUE",
};
exports.SNOMEDCTTraitName = {
    DIAGNOSIS: "DIAGNOSIS",
    FUTURE: "FUTURE",
    HYPOTHETICAL: "HYPOTHETICAL",
    LOW_CONFIDENCE: "LOW_CONFIDENCE",
    NEGATION: "NEGATION",
    PAST_HISTORY: "PAST_HISTORY",
    PERTAINS_TO_FAMILY: "PERTAINS_TO_FAMILY",
    SIGN: "SIGN",
    SYMPTOM: "SYMPTOM",
};
exports.SNOMEDCTAttributeType = {
    ACUITY: "ACUITY",
    DIRECTION: "DIRECTION",
    QUALITY: "QUALITY",
    SYSTEM_ORGAN_SITE: "SYSTEM_ORGAN_SITE",
    TEST_UNIT: "TEST_UNIT",
    TEST_VALUE: "TEST_VALUE",
};
exports.SNOMEDCTEntityType = {
    DX_NAME: "DX_NAME",
    PROCEDURE_NAME: "PROCEDURE_NAME",
    TEST_NAME: "TEST_NAME",
    TREATMENT_NAME: "TREATMENT_NAME",
};
class ValidationException extends ComprehendMedicalServiceException_1.ComprehendMedicalServiceException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
exports.ValidationException = ValidationException;
