// @aws-sdk/client-comprehendmedical/dist-cjs/protocols/Aws_json1_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.de_StopPHIDetectionJobCommand = exports.de_StopICD10CMInferenceJobCommand = exports.de_StopEntitiesDetectionV2JobCommand = exports.de_StartSNOMEDCTInferenceJobCommand = exports.de_StartRxNormInferenceJobCommand = exports.de_StartPHIDetectionJobCommand = exports.de_StartICD10CMInferenceJobCommand = exports.de_StartEntitiesDetectionV2JobCommand = exports.de_ListSNOMEDCTInferenceJobsCommand = exports.de_ListRxNormInferenceJobsCommand = exports.de_ListPHIDetectionJobsCommand = exports.de_ListICD10CMInferenceJobsCommand = exports.de_ListEntitiesDetectionV2JobsCommand = exports.de_InferSNOMEDCTCommand = exports.de_InferRxNormCommand = exports.de_InferICD10CMCommand = exports.de_DetectPHICommand = exports.de_DetectEntitiesV2Command = exports.de_DetectEntitiesCommand = exports.de_DescribeSNOMEDCTInferenceJobCommand = exports.de_DescribeRxNormInferenceJobCommand = exports.de_DescribePHIDetectionJobCommand = exports.de_DescribeICD10CMInferenceJobCommand = exports.de_DescribeEntitiesDetectionV2JobCommand = exports.se_StopSNOMEDCTInferenceJobCommand = exports.se_StopRxNormInferenceJobCommand = exports.se_StopPHIDetectionJobCommand = exports.se_StopICD10CMInferenceJobCommand = exports.se_StopEntitiesDetectionV2JobCommand = exports.se_StartSNOMEDCTInferenceJobCommand = exports.se_StartRxNormInferenceJobCommand = exports.se_StartPHIDetectionJobCommand = exports.se_StartICD10CMInferenceJobCommand = exports.se_StartEntitiesDetectionV2JobCommand = exports.se_ListSNOMEDCTInferenceJobsCommand = exports.se_ListRxNormInferenceJobsCommand = exports.se_ListPHIDetectionJobsCommand = exports.se_ListICD10CMInferenceJobsCommand = exports.se_ListEntitiesDetectionV2JobsCommand = exports.se_InferSNOMEDCTCommand = exports.se_InferRxNormCommand = exports.se_InferICD10CMCommand = exports.se_DetectPHICommand = exports.se_DetectEntitiesV2Command = exports.se_DetectEntitiesCommand = exports.se_DescribeSNOMEDCTInferenceJobCommand = exports.se_DescribeRxNormInferenceJobCommand = exports.se_DescribePHIDetectionJobCommand = exports.se_DescribeICD10CMInferenceJobCommand = exports.se_DescribeEntitiesDetectionV2JobCommand = void 0;
exports.de_StopSNOMEDCTInferenceJobCommand = exports.de_StopRxNormInferenceJobCommand = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const protocol_http_1 = require("@smithy/protocol-http");
const uuid_1 = require("uuid");
const ComprehendMedicalServiceException_1 = require("../models/ComprehendMedicalServiceException");
const models_0_1 = require("../models/models_0");
const se_DescribeEntitiesDetectionV2JobCommand = async (input, context) => {
    const headers = sharedHeaders("DescribeEntitiesDetectionV2Job");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DescribeEntitiesDetectionV2JobCommand = se_DescribeEntitiesDetectionV2JobCommand;
const se_DescribeICD10CMInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("DescribeICD10CMInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DescribeICD10CMInferenceJobCommand = se_DescribeICD10CMInferenceJobCommand;
const se_DescribePHIDetectionJobCommand = async (input, context) => {
    const headers = sharedHeaders("DescribePHIDetectionJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DescribePHIDetectionJobCommand = se_DescribePHIDetectionJobCommand;
const se_DescribeRxNormInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("DescribeRxNormInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DescribeRxNormInferenceJobCommand = se_DescribeRxNormInferenceJobCommand;
const se_DescribeSNOMEDCTInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("DescribeSNOMEDCTInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DescribeSNOMEDCTInferenceJobCommand = se_DescribeSNOMEDCTInferenceJobCommand;
const se_DetectEntitiesCommand = async (input, context) => {
    const headers = sharedHeaders("DetectEntities");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DetectEntitiesCommand = se_DetectEntitiesCommand;
const se_DetectEntitiesV2Command = async (input, context) => {
    const headers = sharedHeaders("DetectEntitiesV2");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DetectEntitiesV2Command = se_DetectEntitiesV2Command;
const se_DetectPHICommand = async (input, context) => {
    const headers = sharedHeaders("DetectPHI");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_DetectPHICommand = se_DetectPHICommand;
const se_InferICD10CMCommand = async (input, context) => {
    const headers = sharedHeaders("InferICD10CM");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_InferICD10CMCommand = se_InferICD10CMCommand;
const se_InferRxNormCommand = async (input, context) => {
    const headers = sharedHeaders("InferRxNorm");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_InferRxNormCommand = se_InferRxNormCommand;
const se_InferSNOMEDCTCommand = async (input, context) => {
    const headers = sharedHeaders("InferSNOMEDCT");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_InferSNOMEDCTCommand = se_InferSNOMEDCTCommand;
const se_ListEntitiesDetectionV2JobsCommand = async (input, context) => {
    const headers = sharedHeaders("ListEntitiesDetectionV2Jobs");
    let body;
    body = JSON.stringify(se_ListEntitiesDetectionV2JobsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_ListEntitiesDetectionV2JobsCommand = se_ListEntitiesDetectionV2JobsCommand;
const se_ListICD10CMInferenceJobsCommand = async (input, context) => {
    const headers = sharedHeaders("ListICD10CMInferenceJobs");
    let body;
    body = JSON.stringify(se_ListICD10CMInferenceJobsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_ListICD10CMInferenceJobsCommand = se_ListICD10CMInferenceJobsCommand;
const se_ListPHIDetectionJobsCommand = async (input, context) => {
    const headers = sharedHeaders("ListPHIDetectionJobs");
    let body;
    body = JSON.stringify(se_ListPHIDetectionJobsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_ListPHIDetectionJobsCommand = se_ListPHIDetectionJobsCommand;
const se_ListRxNormInferenceJobsCommand = async (input, context) => {
    const headers = sharedHeaders("ListRxNormInferenceJobs");
    let body;
    body = JSON.stringify(se_ListRxNormInferenceJobsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_ListRxNormInferenceJobsCommand = se_ListRxNormInferenceJobsCommand;
const se_ListSNOMEDCTInferenceJobsCommand = async (input, context) => {
    const headers = sharedHeaders("ListSNOMEDCTInferenceJobs");
    let body;
    body = JSON.stringify(se_ListSNOMEDCTInferenceJobsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_ListSNOMEDCTInferenceJobsCommand = se_ListSNOMEDCTInferenceJobsCommand;
const se_StartEntitiesDetectionV2JobCommand = async (input, context) => {
    const headers = sharedHeaders("StartEntitiesDetectionV2Job");
    let body;
    body = JSON.stringify(se_StartEntitiesDetectionV2JobRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StartEntitiesDetectionV2JobCommand = se_StartEntitiesDetectionV2JobCommand;
const se_StartICD10CMInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StartICD10CMInferenceJob");
    let body;
    body = JSON.stringify(se_StartICD10CMInferenceJobRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StartICD10CMInferenceJobCommand = se_StartICD10CMInferenceJobCommand;
const se_StartPHIDetectionJobCommand = async (input, context) => {
    const headers = sharedHeaders("StartPHIDetectionJob");
    let body;
    body = JSON.stringify(se_StartPHIDetectionJobRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StartPHIDetectionJobCommand = se_StartPHIDetectionJobCommand;
const se_StartRxNormInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StartRxNormInferenceJob");
    let body;
    body = JSON.stringify(se_StartRxNormInferenceJobRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StartRxNormInferenceJobCommand = se_StartRxNormInferenceJobCommand;
const se_StartSNOMEDCTInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StartSNOMEDCTInferenceJob");
    let body;
    body = JSON.stringify(se_StartSNOMEDCTInferenceJobRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StartSNOMEDCTInferenceJobCommand = se_StartSNOMEDCTInferenceJobCommand;
const se_StopEntitiesDetectionV2JobCommand = async (input, context) => {
    const headers = sharedHeaders("StopEntitiesDetectionV2Job");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StopEntitiesDetectionV2JobCommand = se_StopEntitiesDetectionV2JobCommand;
const se_StopICD10CMInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StopICD10CMInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StopICD10CMInferenceJobCommand = se_StopICD10CMInferenceJobCommand;
const se_StopPHIDetectionJobCommand = async (input, context) => {
    const headers = sharedHeaders("StopPHIDetectionJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StopPHIDetectionJobCommand = se_StopPHIDetectionJobCommand;
const se_StopRxNormInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StopRxNormInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StopRxNormInferenceJobCommand = se_StopRxNormInferenceJobCommand;
const se_StopSNOMEDCTInferenceJobCommand = async (input, context) => {
    const headers = sharedHeaders("StopSNOMEDCTInferenceJob");
    let body;
    body = JSON.stringify((0, smithy_client_1._json)(input));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.se_StopSNOMEDCTInferenceJobCommand = se_StopSNOMEDCTInferenceJobCommand;
const de_DescribeEntitiesDetectionV2JobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DescribeEntitiesDetectionV2JobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DescribeEntitiesDetectionV2JobResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DescribeEntitiesDetectionV2JobCommand = de_DescribeEntitiesDetectionV2JobCommand;
const de_DescribeEntitiesDetectionV2JobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DescribeICD10CMInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DescribeICD10CMInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DescribeICD10CMInferenceJobResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DescribeICD10CMInferenceJobCommand = de_DescribeICD10CMInferenceJobCommand;
const de_DescribeICD10CMInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DescribePHIDetectionJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DescribePHIDetectionJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DescribePHIDetectionJobResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DescribePHIDetectionJobCommand = de_DescribePHIDetectionJobCommand;
const de_DescribePHIDetectionJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DescribeRxNormInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DescribeRxNormInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DescribeRxNormInferenceJobResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DescribeRxNormInferenceJobCommand = de_DescribeRxNormInferenceJobCommand;
const de_DescribeRxNormInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DescribeSNOMEDCTInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DescribeSNOMEDCTInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DescribeSNOMEDCTInferenceJobResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DescribeSNOMEDCTInferenceJobCommand = de_DescribeSNOMEDCTInferenceJobCommand;
const de_DescribeSNOMEDCTInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DetectEntitiesCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DetectEntitiesCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DetectEntitiesResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DetectEntitiesCommand = de_DetectEntitiesCommand;
const de_DetectEntitiesCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DetectEntitiesV2Command = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DetectEntitiesV2CommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DetectEntitiesV2Response(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DetectEntitiesV2Command = de_DetectEntitiesV2Command;
const de_DetectEntitiesV2CommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_DetectPHICommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_DetectPHICommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_DetectPHIResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_DetectPHICommand = de_DetectPHICommand;
const de_DetectPHICommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_InferICD10CMCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_InferICD10CMCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_InferICD10CMResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_InferICD10CMCommand = de_InferICD10CMCommand;
const de_InferICD10CMCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_InferRxNormCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_InferRxNormCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_InferRxNormResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_InferRxNormCommand = de_InferRxNormCommand;
const de_InferRxNormCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_InferSNOMEDCTCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_InferSNOMEDCTCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_InferSNOMEDCTResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_InferSNOMEDCTCommand = de_InferSNOMEDCTCommand;
const de_InferSNOMEDCTCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidEncodingException":
        case "com.amazonaws.comprehendmedical#InvalidEncodingException":
            throw await de_InvalidEncodingExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ServiceUnavailableException":
        case "com.amazonaws.comprehendmedical#ServiceUnavailableException":
            throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
        case "TextSizeLimitExceededException":
        case "com.amazonaws.comprehendmedical#TextSizeLimitExceededException":
            throw await de_TextSizeLimitExceededExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_ListEntitiesDetectionV2JobsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_ListEntitiesDetectionV2JobsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_ListEntitiesDetectionV2JobsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_ListEntitiesDetectionV2JobsCommand = de_ListEntitiesDetectionV2JobsCommand;
const de_ListEntitiesDetectionV2JobsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.comprehendmedical#ValidationException":
            throw await de_ValidationExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_ListICD10CMInferenceJobsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_ListICD10CMInferenceJobsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_ListICD10CMInferenceJobsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_ListICD10CMInferenceJobsCommand = de_ListICD10CMInferenceJobsCommand;
const de_ListICD10CMInferenceJobsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.comprehendmedical#ValidationException":
            throw await de_ValidationExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_ListPHIDetectionJobsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_ListPHIDetectionJobsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_ListPHIDetectionJobsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_ListPHIDetectionJobsCommand = de_ListPHIDetectionJobsCommand;
const de_ListPHIDetectionJobsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.comprehendmedical#ValidationException":
            throw await de_ValidationExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_ListRxNormInferenceJobsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_ListRxNormInferenceJobsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_ListRxNormInferenceJobsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_ListRxNormInferenceJobsCommand = de_ListRxNormInferenceJobsCommand;
const de_ListRxNormInferenceJobsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.comprehendmedical#ValidationException":
            throw await de_ValidationExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_ListSNOMEDCTInferenceJobsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_ListSNOMEDCTInferenceJobsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = de_ListSNOMEDCTInferenceJobsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_ListSNOMEDCTInferenceJobsCommand = de_ListSNOMEDCTInferenceJobsCommand;
const de_ListSNOMEDCTInferenceJobsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.comprehendmedical#ValidationException":
            throw await de_ValidationExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StartEntitiesDetectionV2JobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StartEntitiesDetectionV2JobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StartEntitiesDetectionV2JobCommand = de_StartEntitiesDetectionV2JobCommand;
const de_StartEntitiesDetectionV2JobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StartICD10CMInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StartICD10CMInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StartICD10CMInferenceJobCommand = de_StartICD10CMInferenceJobCommand;
const de_StartICD10CMInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StartPHIDetectionJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StartPHIDetectionJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StartPHIDetectionJobCommand = de_StartPHIDetectionJobCommand;
const de_StartPHIDetectionJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StartRxNormInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StartRxNormInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StartRxNormInferenceJobCommand = de_StartRxNormInferenceJobCommand;
const de_StartRxNormInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StartSNOMEDCTInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StartSNOMEDCTInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StartSNOMEDCTInferenceJobCommand = de_StartSNOMEDCTInferenceJobCommand;
const de_StartSNOMEDCTInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StopEntitiesDetectionV2JobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StopEntitiesDetectionV2JobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StopEntitiesDetectionV2JobCommand = de_StopEntitiesDetectionV2JobCommand;
const de_StopEntitiesDetectionV2JobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StopICD10CMInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StopICD10CMInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StopICD10CMInferenceJobCommand = de_StopICD10CMInferenceJobCommand;
const de_StopICD10CMInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StopPHIDetectionJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StopPHIDetectionJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StopPHIDetectionJobCommand = de_StopPHIDetectionJobCommand;
const de_StopPHIDetectionJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StopRxNormInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StopRxNormInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StopRxNormInferenceJobCommand = de_StopRxNormInferenceJobCommand;
const de_StopRxNormInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_StopSNOMEDCTInferenceJobCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_StopSNOMEDCTInferenceJobCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = (0, smithy_client_1._json)(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
exports.de_StopSNOMEDCTInferenceJobCommand = de_StopSNOMEDCTInferenceJobCommand;
const de_StopSNOMEDCTInferenceJobCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.comprehendmedical#InternalServerException":
            throw await de_InternalServerExceptionRes(parsedOutput, context);
        case "InvalidRequestException":
        case "com.amazonaws.comprehendmedical#InvalidRequestException":
            throw await de_InvalidRequestExceptionRes(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.comprehendmedical#ResourceNotFoundException":
            throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
        case "TooManyRequestsException":
        case "com.amazonaws.comprehendmedical#TooManyRequestsException":
            throw await de_TooManyRequestsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_InternalServerExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.InternalServerException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_InvalidEncodingExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.InvalidEncodingException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_InvalidRequestExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.InvalidRequestException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_ResourceNotFoundExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.ResourceNotFoundException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_ServiceUnavailableExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.ServiceUnavailableException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_TextSizeLimitExceededExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.TextSizeLimitExceededException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_TooManyRequestsExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.TooManyRequestsException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const de_ValidationExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = (0, smithy_client_1._json)(body);
    const exception = new models_0_1.ValidationException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const se_ComprehendMedicalAsyncJobFilter = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        JobName: [],
        JobStatus: [],
        SubmitTimeAfter: (_) => Math.round(_.getTime() / 1000),
        SubmitTimeBefore: (_) => Math.round(_.getTime() / 1000),
    });
};
const se_ListEntitiesDetectionV2JobsRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        Filter: (_) => se_ComprehendMedicalAsyncJobFilter(_, context),
        MaxResults: [],
        NextToken: [],
    });
};
const se_ListICD10CMInferenceJobsRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        Filter: (_) => se_ComprehendMedicalAsyncJobFilter(_, context),
        MaxResults: [],
        NextToken: [],
    });
};
const se_ListPHIDetectionJobsRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        Filter: (_) => se_ComprehendMedicalAsyncJobFilter(_, context),
        MaxResults: [],
        NextToken: [],
    });
};
const se_ListRxNormInferenceJobsRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        Filter: (_) => se_ComprehendMedicalAsyncJobFilter(_, context),
        MaxResults: [],
        NextToken: [],
    });
};
const se_ListSNOMEDCTInferenceJobsRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        Filter: (_) => se_ComprehendMedicalAsyncJobFilter(_, context),
        MaxResults: [],
        NextToken: [],
    });
};
const se_StartEntitiesDetectionV2JobRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        ClientRequestToken: [true, (_) => _ ?? (0, uuid_1.v4)()],
        DataAccessRoleArn: [],
        InputDataConfig: smithy_client_1._json,
        JobName: [],
        KMSKey: [],
        LanguageCode: [],
        OutputDataConfig: smithy_client_1._json,
    });
};
const se_StartICD10CMInferenceJobRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        ClientRequestToken: [true, (_) => _ ?? (0, uuid_1.v4)()],
        DataAccessRoleArn: [],
        InputDataConfig: smithy_client_1._json,
        JobName: [],
        KMSKey: [],
        LanguageCode: [],
        OutputDataConfig: smithy_client_1._json,
    });
};
const se_StartPHIDetectionJobRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        ClientRequestToken: [true, (_) => _ ?? (0, uuid_1.v4)()],
        DataAccessRoleArn: [],
        InputDataConfig: smithy_client_1._json,
        JobName: [],
        KMSKey: [],
        LanguageCode: [],
        OutputDataConfig: smithy_client_1._json,
    });
};
const se_StartRxNormInferenceJobRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        ClientRequestToken: [true, (_) => _ ?? (0, uuid_1.v4)()],
        DataAccessRoleArn: [],
        InputDataConfig: smithy_client_1._json,
        JobName: [],
        KMSKey: [],
        LanguageCode: [],
        OutputDataConfig: smithy_client_1._json,
    });
};
const se_StartSNOMEDCTInferenceJobRequest = (input, context) => {
    return (0, smithy_client_1.take)(input, {
        ClientRequestToken: [true, (_) => _ ?? (0, uuid_1.v4)()],
        DataAccessRoleArn: [],
        InputDataConfig: smithy_client_1._json,
        JobName: [],
        KMSKey: [],
        LanguageCode: [],
        OutputDataConfig: smithy_client_1._json,
    });
};
const de_Attribute = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        RelationshipScore: smithy_client_1.limitedParseFloat32,
        RelationshipType: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_TraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_AttributeList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_Attribute(entry, context);
    });
    return retVal;
};
const de_ComprehendMedicalAsyncJobProperties = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        DataAccessRoleArn: smithy_client_1.expectString,
        EndTime: (_) => (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(_))),
        ExpirationTime: (_) => (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(_))),
        InputDataConfig: smithy_client_1._json,
        JobId: smithy_client_1.expectString,
        JobName: smithy_client_1.expectString,
        JobStatus: smithy_client_1.expectString,
        KMSKey: smithy_client_1.expectString,
        LanguageCode: smithy_client_1.expectString,
        ManifestFilePath: smithy_client_1.expectString,
        Message: smithy_client_1.expectString,
        ModelVersion: smithy_client_1.expectString,
        OutputDataConfig: smithy_client_1._json,
        SubmitTime: (_) => (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(_))),
    });
};
const de_ComprehendMedicalAsyncJobPropertiesList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_ComprehendMedicalAsyncJobProperties(entry, context);
    });
    return retVal;
};
const de_DescribeEntitiesDetectionV2JobResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobProperties: (_) => de_ComprehendMedicalAsyncJobProperties(_, context),
    });
};
const de_DescribeICD10CMInferenceJobResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobProperties: (_) => de_ComprehendMedicalAsyncJobProperties(_, context),
    });
};
const de_DescribePHIDetectionJobResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobProperties: (_) => de_ComprehendMedicalAsyncJobProperties(_, context),
    });
};
const de_DescribeRxNormInferenceJobResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobProperties: (_) => de_ComprehendMedicalAsyncJobProperties(_, context),
    });
};
const de_DescribeSNOMEDCTInferenceJobResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobProperties: (_) => de_ComprehendMedicalAsyncJobProperties(_, context),
    });
};
const de_DetectEntitiesResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Entities: (_) => de_EntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
        UnmappedAttributes: (_) => de_UnmappedAttributeList(_, context),
    });
};
const de_DetectEntitiesV2Response = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Entities: (_) => de_EntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
        UnmappedAttributes: (_) => de_UnmappedAttributeList(_, context),
    });
};
const de_DetectPHIResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Entities: (_) => de_EntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
    });
};
const de_Entity = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Attributes: (_) => de_AttributeList(_, context),
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_TraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_EntityList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_Entity(entry, context);
    });
    return retVal;
};
const de_ICD10CMAttribute = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        RelationshipScore: smithy_client_1.limitedParseFloat32,
        RelationshipType: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_ICD10CMTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_ICD10CMAttributeList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_ICD10CMAttribute(entry, context);
    });
    return retVal;
};
const de_ICD10CMConcept = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Code: smithy_client_1.expectString,
        Description: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_ICD10CMConceptList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_ICD10CMConcept(entry, context);
    });
    return retVal;
};
const de_ICD10CMEntity = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Attributes: (_) => de_ICD10CMAttributeList(_, context),
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        ICD10CMConcepts: (_) => de_ICD10CMConceptList(_, context),
        Id: smithy_client_1.expectInt32,
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_ICD10CMTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_ICD10CMEntityList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_ICD10CMEntity(entry, context);
    });
    return retVal;
};
const de_ICD10CMTrait = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Name: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_ICD10CMTraitList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_ICD10CMTrait(entry, context);
    });
    return retVal;
};
const de_InferICD10CMResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Entities: (_) => de_ICD10CMEntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
    });
};
const de_InferRxNormResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Entities: (_) => de_RxNormEntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
    });
};
const de_InferSNOMEDCTResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Characters: smithy_client_1._json,
        Entities: (_) => de_SNOMEDCTEntityList(_, context),
        ModelVersion: smithy_client_1.expectString,
        PaginationToken: smithy_client_1.expectString,
        SNOMEDCTDetails: smithy_client_1._json,
    });
};
const de_ListEntitiesDetectionV2JobsResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobPropertiesList: (_) => de_ComprehendMedicalAsyncJobPropertiesList(_, context),
        NextToken: smithy_client_1.expectString,
    });
};
const de_ListICD10CMInferenceJobsResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobPropertiesList: (_) => de_ComprehendMedicalAsyncJobPropertiesList(_, context),
        NextToken: smithy_client_1.expectString,
    });
};
const de_ListPHIDetectionJobsResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobPropertiesList: (_) => de_ComprehendMedicalAsyncJobPropertiesList(_, context),
        NextToken: smithy_client_1.expectString,
    });
};
const de_ListRxNormInferenceJobsResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobPropertiesList: (_) => de_ComprehendMedicalAsyncJobPropertiesList(_, context),
        NextToken: smithy_client_1.expectString,
    });
};
const de_ListSNOMEDCTInferenceJobsResponse = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        ComprehendMedicalAsyncJobPropertiesList: (_) => de_ComprehendMedicalAsyncJobPropertiesList(_, context),
        NextToken: smithy_client_1.expectString,
    });
};
const de_RxNormAttribute = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        BeginOffset: smithy_client_1.expectInt32,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        RelationshipScore: smithy_client_1.limitedParseFloat32,
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_RxNormTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_RxNormAttributeList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_RxNormAttribute(entry, context);
    });
    return retVal;
};
const de_RxNormConcept = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Code: smithy_client_1.expectString,
        Description: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_RxNormConceptList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_RxNormConcept(entry, context);
    });
    return retVal;
};
const de_RxNormEntity = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Attributes: (_) => de_RxNormAttributeList(_, context),
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        RxNormConcepts: (_) => de_RxNormConceptList(_, context),
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_RxNormTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_RxNormEntityList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_RxNormEntity(entry, context);
    });
    return retVal;
};
const de_RxNormTrait = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Name: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_RxNormTraitList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_RxNormTrait(entry, context);
    });
    return retVal;
};
const de_SNOMEDCTAttribute = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        RelationshipScore: smithy_client_1.limitedParseFloat32,
        RelationshipType: smithy_client_1.expectString,
        SNOMEDCTConcepts: (_) => de_SNOMEDCTConceptList(_, context),
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_SNOMEDCTTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_SNOMEDCTAttributeList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_SNOMEDCTAttribute(entry, context);
    });
    return retVal;
};
const de_SNOMEDCTConcept = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Code: smithy_client_1.expectString,
        Description: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_SNOMEDCTConceptList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_SNOMEDCTConcept(entry, context);
    });
    return retVal;
};
const de_SNOMEDCTEntity = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Attributes: (_) => de_SNOMEDCTAttributeList(_, context),
        BeginOffset: smithy_client_1.expectInt32,
        Category: smithy_client_1.expectString,
        EndOffset: smithy_client_1.expectInt32,
        Id: smithy_client_1.expectInt32,
        SNOMEDCTConcepts: (_) => de_SNOMEDCTConceptList(_, context),
        Score: smithy_client_1.limitedParseFloat32,
        Text: smithy_client_1.expectString,
        Traits: (_) => de_SNOMEDCTTraitList(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_SNOMEDCTEntityList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_SNOMEDCTEntity(entry, context);
    });
    return retVal;
};
const de_SNOMEDCTTrait = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Name: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_SNOMEDCTTraitList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_SNOMEDCTTrait(entry, context);
    });
    return retVal;
};
const de_Trait = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Name: smithy_client_1.expectString,
        Score: smithy_client_1.limitedParseFloat32,
    });
};
const de_TraitList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_Trait(entry, context);
    });
    return retVal;
};
const de_UnmappedAttribute = (output, context) => {
    return (0, smithy_client_1.take)(output, {
        Attribute: (_) => de_Attribute(_, context),
        Type: smithy_client_1.expectString,
    });
};
const de_UnmappedAttributeList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        return de_UnmappedAttribute(entry, context);
    });
    return retVal;
};
const deserializeMetadata = (output) => ({
    httpStatusCode: output.statusCode,
    requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
    extendedRequestId: output.headers["x-amz-id-2"],
    cfId: output.headers["x-amz-cf-id"],
});
const collectBody = (streamBody = new Uint8Array(), context) => {
    if (streamBody instanceof Uint8Array) {
        return Promise.resolve(streamBody);
    }
    return context.streamCollector(streamBody) || Promise.resolve(new Uint8Array());
};
const collectBodyString = (streamBody, context) => collectBody(streamBody, context).then((body) => context.utf8Encoder(body));
const throwDefaultError = (0, smithy_client_1.withBaseException)(ComprehendMedicalServiceException_1.ComprehendMedicalServiceException);
const buildHttpRpcRequest = async (context, headers, path, resolvedHostname, body) => {
    const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
    const contents = {
        protocol,
        hostname,
        port,
        method: "POST",
        path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
        headers,
    };
    if (resolvedHostname !== undefined) {
        contents.hostname = resolvedHostname;
    }
    if (body !== undefined) {
        contents.body = body;
    }
    return new protocol_http_1.HttpRequest(contents);
};
function sharedHeaders(operation) {
    return {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": `ComprehendMedical_20181030.${operation}`,
    };
}
const parseBody = (streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
    if (encoded.length) {
        return JSON.parse(encoded);
    }
    return {};
});
const parseErrorBody = async (errorBody, context) => {
    const value = await parseBody(errorBody, context);
    value.message = value.message ?? value.Message;
    return value;
};
const loadRestJsonErrorCode = (output, data) => {
    const findKey = (object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase());
    const sanitizeErrorCode = (rawValue) => {
        let cleanValue = rawValue;
        if (typeof cleanValue === "number") {
            cleanValue = cleanValue.toString();
        }
        if (cleanValue.indexOf(",") >= 0) {
            cleanValue = cleanValue.split(",")[0];
        }
        if (cleanValue.indexOf(":") >= 0) {
            cleanValue = cleanValue.split(":")[0];
        }
        if (cleanValue.indexOf("#") >= 0) {
            cleanValue = cleanValue.split("#")[1];
        }
        return cleanValue;
    };
    const headerKey = findKey(output.headers, "x-amzn-errortype");
    if (headerKey !== undefined) {
        return sanitizeErrorCode(output.headers[headerKey]);
    }
    if (data.code !== undefined) {
        return sanitizeErrorCode(data.code);
    }
    if (data["__type"] !== undefined) {
        return sanitizeErrorCode(data["__type"]);
    }
};
